File: 준영/프로그래머스/LV2/카펫.js
function solution(brown, yellow) {
  var answer = []
  var total = brown + yellow
  var idx = total
  while (idx > 0) {
    if (total % idx == 0) {
      var x = idx
      var y = total / idx
      if ((x + y) * 2 - 4 === brown) {
        answer.push(x)
        answer.push(y)
        break
      }
    }
    idx -= 1
  }
  return answer
}

File: 준영/프로그래머스/LV2/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    idx = total
    while idx > 0:
        if total % idx == 0:
            x = (total // idx)
            y = idx
            if x * y == total and (x + y)*2 - 4 == brown:
                answer.append(y)
                answer.append(x)
                break
        idx -= 1
            
    return answer